### PR TITLE
Feat: Add finer control over headline depth.

### PR DESCRIPTION
**Pull request by MediosZ**
This PR allows users to set the depth of headlines displayed,  so that the users don't have to choose between zero and infinity.

By default, all headlines are displayed, keeping the behavior consistent.

### CODE REVIEW

**Comment by turulix** [comment]
The indent level here, means how far each line is indented and does not "filter out" something like h6 or h5, if this was the intention. As seen in this Screenshot, where Test1 should still be shown.
image

 Maybe we could rename this to "Recursion Limit", with it being unlimited by default.
 
 If you want to filter out h6 / h5 and so on, you would need to check
https://github.com/turulix/obsidian-folder-index/blob/cedc1b400ff25d046671491b102c7df4fb4b9dee/src/types/MarkdownTextRenderer.ts#L74C1-L81

if the `headerWrapper.header.level` is <= The requested header depth

File: src/models/PluginSettingsTab.ts [hunk 1]
```diff
@@ -176,6 +178,28 @@ export class PluginSettingsTab extends PluginSettingTab {
 					await this.plugin.saveSettings()
 				}))
 
+		new Setting(containerEl)
+			.setName("Headline Depth")
```
Comment (turulix) [inline src/models/PluginSettingsTab.ts:182]: Rename to Recursion Depth, maybe?

File: src/types/MarkdownTextRenderer.ts [hunk 2]
```diff
@@ -38,7 +38,9 @@ export class MarkdownTextRenderer {
 				} else {
 					markdownText += this.buildMarkdownLinkString(file.name, null, indentLevel, true)
 				}
-				markdownText += this.buildStructureMarkdownText(this.buildFileTree(children), indentLevel + 1)
+				if (indentLevel < this.plugin.settings.headlineDepth) {
```
Comment (turulix) [inline src/types/MarkdownTextRenderer.ts:41]: Filters out any things indented more than `headerlineDepth` which does not nessesarily equal the header level

**Review by turulix — CHANGES_REQUESTED** [review]

**Comment by turulix** [comment]
You need to enable 
image

**Comment by MediosZ** [comment]
Yeah, I didn't consider the recursive sub folders...
I will try to solve this!

**Comment by MediosZ** [comment]
Since we are showing folders and headlines at the same time, maybe it's good to have two limits for folders and headlines.
Then the users will be able to control the depth of the folders and the depth of the headlines separately.

**Comment by turulix** [comment]
Sounds like a good idea to me :)

**Comment by MediosZ** [comment]
I think it works now 😄

**Comment by turulix** [comment]
Currently sick, will deal with this once im healthy again

**Comment by MediosZ** [comment]
Sorry to hear that, hope you get well soon!

**Review by turulix — APPROVED** [review]

**Pull request merged**